lib/matplotlib/axes/_base.py

```diff
@@ -163,31 +163,31 @@ def _process_plot_format(fmt):
     except ValueError:
         pass  # No, not just a color.
 
+    errfmt = "{!r} is not a valid format string ({})"
+    if ambiguous_fmt_datakey:
+        errfmt += "; it is not a valid data key either"
```

Comment:
```suggestion
    if ambiguous_fmt_datakey:
        errfmt = "{!r} is neither a data key nor a valid format string ({})"
    else:
        errfmt = "{!r} is not a valid format string ({})"
```

I assume that if `data` is given, most of such errors are actually misspelled names and not format errors, so let's put the key error first instead of just an afterthought.

Reply:
Sure.